Workflow-Controller panic: can't create an event with namespace 'default' in namespace 'argo''

## Summary
I've noticed a unexpected restart of the pod

What happened/what you expected to happen?
This is instance managed by ArgoCD instance and currently not being actively used

What version is it broken in?
v3.2.6
What version was it working in?

## Diagnostics

What Kubernetes provider are you using?
GKE on 1.21.5-gke.1802

What executor are you running? Docker/K8SAPI/Kubelet/PNS/Emissary

Emissary 
```bash
...
time="2022-01-21T07:54:25.370Z" level=info msg="Update leases 200"
time="2022-01-21T07:54:29.124Z" level=info msg="List workflows 200"
time="2022-01-21T07:54:29.124Z" level=info msg=healthz age=5m0s err="<nil>" instanceID= labelSelector="!workflows.argoproj.io/phase,!workflows.argoproj.io/controller-instanceid" managedNamespace=
time="2022-01-21T07:54:30.377Z" level=info msg="Get leases 200"
time="2022-01-21T07:54:30.385Z" level=info msg="Update leases 200"
time="2022-01-21T07:54:35.431Z" level=info msg="Get leases 200"
time="2022-01-21T07:54:35.442Z" level=info msg="Update leases 200"
time="2022-01-21T07:54:48.550Z" level=info msg="Get leases 500"
E0121 07:54:48.551558       1 leaderelection.go:325] error retrieving resource lock argo/workflow-controller: etcdserver: request timed out
I0121 07:54:50.443189       1 leaderelection.go:278] failed to renew lease argo/workflow-controller: timed out waiting for the condition
E0121 07:54:50.443268       1 leaderelection.go:301] Failed to release lock: resource name may not be empty
time="2022-01-21T07:54:50.443Z" level=info msg="stopped leading" id=argo-workflow-controller-7797b9c9d5-p48p5
E0121 07:54:50.443353       1 event.go:273] Unable to write event: '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:".16cc39f57309758c", GenerateName:"", Namespace:"default", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, InvolvedObject:v1.ObjectReference{Kind:"Lease", Namespace:"", Name:"", UID:"", APIVersion:"coordination.k8s.io/v1", ResourceVersion:"", FieldPath:""}, Reason:"LeaderElection", Message:"argo-workflow-controller-7797b9c9d5-p48p5 stopped leading", Source:v1.EventSource{Component:"workflow-controller", Host:""}, FirstTimestamp:v1.Time{Time:time.Time{wall:0xc072b7929a6a118c, ext:542583113832237, loc:(*time.Location)(0x3346660)}}, LastTimestamp:v1.Time{Time:time.Time{wall:0xc072b7929a6a118c, ext:542583113832237, loc:(*time.Location)(0x3346660)}}, Count:1, Type:"Normal", EventTime:v1.MicroTime{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, Series:(*v1.EventSeries)(nil), Action:"", Related:(*v1.ObjectReference)(nil), ReportingController:"", ReportingInstance:""}': 'can't create an event with namespace 'default' in namespace 'argo''(may retry after sleeping)
panic: http: Server closed

goroutine 148 [running]:
github.com/argoproj/argo-workflows/v3/workflow/metrics.runServer.func1(0x1, 0x21eabe1, 0x8, 0x2382, 0x0, 0x0, 0xc0007bc0e0)
	/go/src/github.com/argoproj/argo-workflows/workflow/metrics/server.go:56 +0x11d
created by github.com/argoproj/argo-workflows/v3/workflow/metrics.runServer
	/go/src/github.com/argoproj/argo-workflows/workflow/metrics/server.go:53 +0x27e
```

---
<!-- Issue Author: Don't delete this message to encourage other users to support your issue! -->
**Message from the maintainers**:

Impacted by this regression? Give it a 👍. We prioritise the issues with the most 👍.

Thanks @alexec!

```
I0224 06:27:16.729930       1 leaderelection.go:278] failed to renew lease kubeflow/workflow-controller: timed out waiting for the condition
E0224 06:27:16.730011       1 leaderelection.go:301] Failed to release lock: resource name may not be empty
time="2022-02-24T06:27:16.730Z" level=info msg="stopped leading" id=workflow-controller-f78bdfff-8ntb2
E0224 06:27:16.730072       1 event.go:273] Unable to write event: 'can't create an event with namespace 'default' in namespace 'kubeflow'' (may retry after sleeping)
time="2022-02-24T06:27:16.730Z" level=info msg="Shutting workflow TTL worker"
panic: http: Server closed

goroutine 235 [running]:
github.com/argoproj/argo-workflows/v3/workflow/metrics.runServer.func1(0x1, 0x1eeb199, 0x8, 0x2382, 0x0, 0x0, 0xc00aa43500)
	/go/src/github.com/argoproj/argo-workflows/workflow/metrics/server.go:56 +0x117
created by github.com/argoproj/argo-workflows/v3/workflow/metrics.runServer
	/go/src/github.com/argoproj/argo-workflows/workflow/metrics/server.go:53 +0x27e
```

--- a/USERS.md
+++ b/USERS.md
@@ -97,6 +97,7 @@ Currently, the following organizations are **officially** using Argo Workflows:
 1. [Localytics](https://www.localytics.com/)
 1. [Lumin Digital](https://lumindigital.com/)
 1. [Maersk](https://www.maersk.com/solutions/digital-solutions)
+1. [MariaDB](https://mariadb.com/)
 1. [Marmalade](https://www.marmalade.co/)
 1. [Max Kelsen](https://maxkelsen.com/)
 1. [Microba](https://www.microba.com/)

--- a/workflow/controller/controller.go
+++ b/workflow/controller/controller.go
@@ -292,7 +292,7 @@ func (wfc *WorkflowController) Run(ctx context.Context, wfWorkers, workflowTTLWo
 				LeaseMeta: metav1.ObjectMeta{Name: leaderName, Namespace: wfc.namespace}, Client: wfc.kubeclientset.CoordinationV1(),
 				LockConfig: resourcelock.ResourceLockConfig{Identity: nodeID, EventRecorder: wfc.eventRecorderManager.Get(wfc.namespace)},
 			},
-			ReleaseOnCancel: true,
+			ReleaseOnCancel: false,
 			LeaseDuration:   env.LookupEnvDurationOr("LEADER_ELECTION_LEASE_DURATION", 15*time.Second),
 			RenewDeadline:   env.LookupEnvDurationOr("LEADER_ELECTION_RENEW_DEADLINE", 10*time.Second),
 			RetryPeriod:     env.LookupEnvDurationOr("LEADER_ELECTION_RETRY_PERIOD", 5*time.Second),

